CIPs/0008.md

```diff
@@ -0,0 +1,77 @@
+# CIP 0008: Expand Metadata to general off-chain storage
+
+- Date: 2020-05-05
+- Author: @nambrot
+- Status: WIP
+
+## Overview
+
+In CIP3, the Celo protocol has started with a simple off-chain storage mechanism mainly focused on claims an account can make about their off-chain identity and communicate to other participants in the network. Over time, there have been additional use cases which have similar access patterns that prompted the creation of `Metadata` in the first place. I.e. data that does not require the expensive cost of "hard consensus", instead just needs to be authenticated by a single party.
+
+Examples include:
+- Transaction metadata such as comments or encrypted messaging in general
+- Profile information such as name or a profile picture
+- Unidirectional attestations like an audit or badges won in The Great Celo Stake Off
+- Signed statements/claims by a party for relase binaries/docker images/package hashes
+
+## Goals
+
+- Expand storage and indexing capabilities of this off-chain data
+- Encode the notion of decryption keys
+- Lightweight messaging system
+
+
+## Proposed Solution
+
+We propose to add another claim, called the `STORAGE` claim:
+```json
+{
+  "type": "STORAGE",
+  "timestamp": 1588886114907,
+  "address": "https://example.com/celo_storage",
+  "filterdataPaths": ".*"
```

Comment:
This has a nasty sharp edge that will allow for DoS attacks against clients that attempt to evaluate this regex. I would suggest using a simpler glob scheme, or at least adding a warning here https://en.wikipedia.org/wiki/ReDoS

Reply:
Updated it to use a glob scheme like https://github.com/isaacs/minimatch